apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -138,4 +158,24 @@ defmodule GameBackend.Matches do
 
   defp get_operation_result({:ok, _}, {:ok, _}), do: {:ok, nil}
   defp get_operation_result(_, _), do: {:error, nil}
+
+  defp complete_quest_and_insert_currency(daily_quest, user_id) do
+    updated_match =
+      UserQuest.changeset(daily_quest, %{
+        completed: true,
+        completed_at: DateTime.utc_now(),
+        status: "completed"
+      })
+      |> Repo.update()
+
+    inserted_currency =
+      Currencies.add_currency_by_name_and_game(
+        user_id,
+        daily_quest.quest.reward["currency"],
+        Utils.get_game_id(:curse_of_mirra),
+        daily_quest.quest.reward["amount"]
+      )
+
+    get_operation_result(updated_match, inserted_currency)
```

Comment:
This is complicated for a reason, we need access to the different results from previous steps in the multi, for example in the daily quest operations we need access to the google users, and to avoid doing a lot of queries we just fetch that once and just use it in the nexts steps. The same go for the bounties when we get the inserted results `Map.get(changes_so_far, {:insert, result["user_id"]})`